File: 1894-merge-strings-alternately/merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        i, j = 0, 0
        n, m = len(word1), len(word2)
        result = []

        while i < n and j < m:
            result.append(word1[i])
            result.append(word2[j])
            i += 1
            j += 1

        if i < n:
            result.append(word1[i:])
        if j < m:
            result.append(word2[j:])

        return ''.join(result)

File: 1894-merge-strings-alternately/merge-strings-alternately.cs
public class Solution {
    public string MergeAlternately(string word1, string word2) {
        var merged = new System.Text.StringBuilder();
        int i = 0, j = 0;
        int len1 = word1.Length, len2 = word2.Length;

        while (i < len1 || j < len2) {
            if (i < len1) merged.Append(word1[i++]);
            if (j < len2) merged.Append(word2[j++]);
        }

        return merged.ToString();
    }
}